enkit/outputs/commands.go

```diff
@@ -109,10 +107,18 @@ func NewMount(root *Root) *Mount {
 }
 
 // maybeSetupTunnel takes a "host:port" string and starts a background tunnel
-// targeting that host/port if necessary. It returns a host and port that
+// if necessary. It returns a host and port that
 // clients should connect to, which could either be the original host/port if no
 // tunnel was necessary, or a modified host/port if a tunnel was necessary.
-func maybeSetupTunnel(hostPort string, tunnelTarget string, gatewayProxy string) (string, int, error) {
+//
+// More specifically, if a tunnel is not necessary, this returns the original
+// host:port (but split into host and port).
+// If a tunnel was necessary, this:
+// * creates the tunnel to tunnelTarget listening on tunnelPort on localhost
+// * returns the original host but changes the port to the tunnelPort
+// It is expected that the supplied host resolve to localhost, so that the
+// connection for the returned host and port combo jumps through the tunnel.
+func maybeSetupTunnel(hostPort string, tunnelTarget string, tunnelPort int, gatewayProxy string) (string, int, error) {
```

Comment:
maybe is kinda weird function name... I cant really think of a better name though.